src/main/java/taxi/controller/driver/AddDriverController.java

```diff
@@ -21,11 +21,22 @@ public void doGet(HttpServletRequest req, HttpServletResponse resp)
     }
 
     @Override
-    public void doPost(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+    public void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws IOException, ServletException {
         String name = req.getParameter("name");
         String licenseNumber = req.getParameter("license_number");
-        Driver driver = new Driver(name, licenseNumber);
+        String login = req.getParameter("login");
+        String password = req.getParameter("password");
+        if (name == null || name.length() == 0
+                || licenseNumber == null || licenseNumber.length() == 0
+                || login == null || login.length() == 0
+                || password == null || password.length() == 0) {
+            req.setAttribute("errorMsg", "Invalid name or country!");
+            req.getRequestDispatcher("/WEB-INF/views/manufacturers/add.jsp").forward(req, resp);
+        }
```

Comment:
same, and fix the error message with redirect

Reply:
I fix it, but it still not working, because if I try to leave an empty line I can't click on the submit button. I get the message: "Заповніть це поле"

Reply:
I am guessing this case was covered on the FE part, but we also need t cover a case when some 3rd party service can use our API and it should be consistent and independent from the UI